inc/features/blocks/personalization/edit.js

```diff
@@ -0,0 +1,351 @@
+	// Ensure variant data is correct parentId is correct.
+	variants.forEach( variant => {
+		if ( ! variant.attributes.parentId || variant.attributes.parentId !== attributes.clientId ) {
+			setVariantAttributes( variant.clientId, {
+				parentId: attributes.clientId,
+			} );
+		}
+	} );
+
+	return (
+		<Fragment>
+			<BlockControls>
+				<Toolbar
+					className="altis-variants-toolbar"
+					controls={ [
+						{
+							icon: 'plus',
+							title: __( 'Add a variant', 'altis-experiments' ),
+							className: 'altis-add-variant-button',
+							onClick: () => {
+								const newClientId = addVariant();
+								setVariant( newClientId );
+							},
+						},
+					] }
+				>
+					{ variants.map( variant => (
+						<Button
+							key={ `variant-select-${ variant.clientId }` }
+							className={ `altis-variant-button components-icon-button has-text ${ activeVariant === variant.clientId && 'is-active' }` }
+							title={ __( 'Select variant', 'altis-experiments' ) }
+							onClick={ () => setVariant( variant.clientId ) }
+						>
+							<VariantTitle variant={ variant } />
+						</Button>
+					) ) }
+				</Toolbar>
+			</BlockControls>
+			<InspectorControls>
+				{ variants.map( variant => {
+					if ( variant.attributes.fallback ) {
+						return (
+							<PanelBody
+								key={ `variant-settings-${ variant.clientId }` }
+								title={ __( 'Fallback', 'altis-experiments' ) }
+							>
+								<p className="description">
+									{ __( 'This variant will be shown as a fallback if no audiences are matched. You can leave the content empty if you do not wish to show anything.', 'altis-experiments' ) }
+								</p>
+							</PanelBody>
+						);
+					}
+
+					return (
+						<PanelBody
+							key={ `variant-settings-${ variant.clientId }` }
+							title={ <VariantTitle variant={ variant } /> }
+						>
+							<AudiencePicker
+								label={ __( 'Audience' ) }
+								audience={ variant.attributes.audience }
+								onSelect={ audience => {
+									setVariantAttributes( variant.clientId, { audience: audience.id } );
+								} }
+								onClearSelection={ () => {
+									setVariantAttributes( variant.clientId, { audience: null } );
+								} }
+							/>
+							{ ! variant.attributes.audience && (
+								<p className="description">
+									{ __( 'You must select an audience for this variant.', 'altis-experiments' ) }
+								</p>
+							) }
+						</PanelBody>
+					);
+				} ) }
+			</InspectorControls>
+			<style dangerouslySetInnerHTML={ {
+				__html: `
+					[data-block="${ clientId }"] [data-type="altis/personalization-variant"] {
+						display: none;
+					}
+					[data-block="${ clientId }"] #block-${ activeVariant } {
+						display: block;
+					}
+				`,
+			} } />
+			<div className={ className }>
+				<div className="wp-core-ui altis-experience-block-header">
+					<span className="altis-experience-block-header__title">
+						{ __( 'Personalized Content', 'altis-experiments' ) }
+						{ ' ・ ' }
```

Comment:
Could handle via CSS?

Reply:
Think I'll leave it for now tbh, there's not really anything to gain from moving this CSS, plus punctuation is content.